spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/ChannelBindingService.java

```diff
@@ -82,4 +120,26 @@ public void unbindConsumers(String inputChannelName) {
 	public void unbindProducers(String outputChannelName) {
 		this.binder.unbindProducers(outputChannelName);
 	}
+
+	/**
+	 * Setup data-type and message converters for the given message channel.
+	 *
+	 * @param messageChannel message channel to set the data-type and message converters
+	 * @param channelName the channel name
+	 */
+	public void configureMessageConverters(MessageChannel messageChannel, String channelName) {
+		Assert.isAssignable(AbstractMessageChannel.class, messageChannel.getClass());
```

Comment:
Given where this is called from, I don't think they'll ever be proxified. But that also hints that we should perhaps consider factoring this class out eventually - too much stuff in the middle right now (although necessary ATM). Checking for a proxy wouldn't hurt though.